pkg/currency/currency.go

```diff
@@ -0,0 +1,42 @@
+package currency
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+// Picodollar is a type to represent currency with 12 decimal precision
+type PicoDollar int64
+
+const (
+	PicoDollarsPerDollar = 1e12
+)
+
+// FromDollars converts a dollar amount (as a float) to Picodollars
+// This should mostly be used for testing, and real usage should be done purely in PicoDollars
+func FromDollars(dollars float64) (PicoDollar, error) {
+	if math.IsNaN(dollars) || math.IsInf(dollars, 0) {
+		return 0, errors.New("invalid dollar amount: must be a finite number")
+	}
+
+	picodollars := dollars * PicoDollarsPerDollar
+	if picodollars > math.MaxInt64 || picodollars < math.MinInt64 {
+		return 0, errors.New("dollar amount too large to convert to Picodollars")
+	}
+	return PicoDollar(picodollars), nil
+}
+
+// ToDollars converts PicoDollars to a dollar amount (as a float)
+func (p PicoDollar) toDollars() float64 {
+	return float64(p) / PicoDollarsPerDollar
+}
```

Comment:
oof. I really hate these float calculations... Can we call it `toDollarsTestOnly` or something? this should not be called in prod code

Reply:
Yes, it should only be used for tests. I can rename to make that more clear.